src/rust/engine/workunit_store/src/lib.rs

```diff
@@ -164,7 +169,12 @@ impl UserMetadataItem {
 
 enum StoreMsg {
   Started(Workunit),
-  Completed(SpanId, Option<WorkunitMetadata>, SystemTime),
+  Completed(
+    SpanId,
+    Option<WorkunitMetadata>,
+    SystemTime,
+    Option<HashMap<Metric, u64>>,
```

Comment:
An empty `HashMap` [doesn't allocate](https://doc.rust-lang.org/std/collections/struct.HashMap.html#method.new), so there is no need to wrap it in an `Option`.